third_party/tcplp/bsdtcp/tcp_subr.c

```diff
@@ -158,6 +169,8 @@ void initialize_tcb(struct tcpcb* tp) {
 void
 tcp_discardcb(struct tcpcb *tp)
 {
+	reinitialize_tcb(tp);
```

Comment:
This placement of the call to `reinitialize_tcb` looks like it might be OK, but I'm wondering if this is better placed at the end of `tcp_discardcb`, or in `tcp_close_tcb` after the call to `tcp_discardcb`. If you place it here, the timers will appear cleared and `cb_destroy` won't be called --- I don't think that's a problem currently, but might be in the future if a different congestion control algorithm is added. If it's something you can test, then it might make sense to try moving this call later.

Reply:
Currently `tcp_discardcb` does not call `cb_destroy` because `CC_ALGO(tp)` is a constant and its `cb_destroy` is initialized to NULL.

`initialize_tcb` does not affect any calls to `cb_destroy` because `struct cc_algo *cc_algo` of `struct tcpcb` is commented out.

It is OK to call `reinitialize_tcb` after `tcp_discardcb`.

However, I noticed the comment `Don't reset the TCB by calling initialize_tcb, since that overwrites the buffer contents.` in `tcp_close_tcb`. Does it have any special meaning?